File: C0120G1/Module1/src/CaseStudy/6.Hướng đối tượng/Customer.js
let Customer = function () {
    this.setNameCustomer = function (nameCustomer) {
        this.nameCustomer = nameCustomer;
    };

    this.getNameCustomer = function () {
        return this.nameCustomer;
    };

    this.setIdCard = function (idCard) {
        this.idCard = idCard;
    };

    this.getIdCard = function () {
        return this.idCard;
    };

    this.setBirthdayCustomer = function (birthdayCustomer) {
        this.birthdayCustomer = birthdayCustomer;
    };

    this.getBirthdayCustomer = function () {
        return this.birthdayCustomer;
    };

    this.setEmailCustomer = function (emailCustomer) {
        this.emailCustomer = emailCustomer;
    };

    this.getEmailCustomer = function () {
        return this.emailCustomer;
    };

    this.setAddressCustomer = function (addressCustomer) {
        this.addressCustomer = addressCustomer;
    };

    this.getAddressCustomer = function () {
        return this.addressCustomer;
    };

    this.setTypeCustomer = function (typeCustomer) {
        this.typeCustomer = typeCustomer;
    };

    this.getTypeCustomer = function () {
        return this.typeCustomer;
    };

    this.setDiscount = function (discount) {
        this.discount = discount;
    };

    this.getDiscount = function () {
        return this.discount;
    };

    this.setNumberOfAccompanying = function (numberOfAccompanying) {
        this.numberOfAccompanying = numberOfAccompanying;
    };

    this.getNumberOfAccompanying = function () {
        return this.numberOfAccompanying;
    };

    this.setTypeRoom = function (typeRoom) {
        this.typeRoom = typeRoom;
    };

    this.getTypeRoom = function () {
        return this.typeRoom;
    };

    this.setRentDays = function (rentDays) {
        this.rentDays = rentDays;
    };

    this.getRentDays = function () {
        return this.rentDays;
    };

    this.setTypeService = function (typeService) {
        this.typeService = typeService;
    };

    this.getTypeService = function () {
        return this.typeService;
    };

    this.totalPays = function () {
        let moneyPerDays=0;
        if(this.getTypeService()=== "Villa"){
            moneyPerDays=500;
        }else if(this.getTypeService()=== "House"){
            moneyPerDays=300;
        }else{
            moneyPerDays=300;
        }
      return moneyPerDays*parseFloat(this.getRentDays())*(1-parseFloat(this.getDiscount())/100);
    }
};
let Employee=function () {
    this.setName=function (name) {
        this.name=name;
    };
    this.getName=function () {
        return this.name;
    };
    this.setBirthday=function (birthday) {
        this.birthday=birthday;
    };
    this.getBirthday=function () {
        return this.birthday;
    };
    this.setIdCard=function (idCard) {
        this.idCard=idCard;
    };
    this.getIdCard=function () {
        return this.idCard;
    };
    this.setPhone=function (phone) {
        this.phone=phone;
    };
    this.getPhone=function () {
        return this.phone;
    };
    this.setEmail=function (email) {
        this.email=email;
    };
    this.getEmail=function () {
        return this.email;
    };
    this.setLevel=function (level) {
        this.level=level;
    };
    this.getLevel=function () {
        return this.level;
    };
    this.setLocal=function (local) {
        this.local=local;
    };
    this.getLocal=function () {
        return this.local;
    };
    this.setSalary=function (salary) {
        this.salary=salary;
    };
    this.getSalary=function () {
        return this.salary;
    };
    this.totalPays = function () {
        let moneyPerDays=0;
        if(this.getTypeService()=== "Villa"){
            moneyPerDays=500;
        }else if(this.getTypeService()=== "House"){
            moneyPerDays=300;
        }else{
            moneyPerDays=300;
        }
        return moneyPerDays*parseFloat(this.getRentDays())*(1-parseFloat(this.getDiscount())/100);
    };
    this.totalSalary=function () {
        let salary=0;
        if(this.getLocal()=="Manager"){
            salary=parseInt(this.getSalary())+500;
        }else if(this.getLocal()=="Sale"){
            salary=parseInt(this.getSalary())+300;
        }else if(this.getLocal()=="Marketing"){
            salary=parseInt(this.getSalary())+200;
        }
        return salary;
    };
}

File: C0120G1/Module2/src/CaseStudy_Module2/Models/Customer.java
package CaseStudy_Module2.Models;

public class Customer {
    private String nameCustomer;
    private String gender;
    private String birthday;
    private int idCard;
    private int phoneNumber;
    private String email;
    private String typeCustomer;
    private String address;
    private Service service;

    public Customer() {
    }

    public Customer(String nameCustomer, String gender, String birthday, int idCard, int phoneNumber, String email, String typeCustomer, String address) {
        this.nameCustomer = nameCustomer;
        this.gender = gender;
        this.birthday = birthday;
        this.idCard = idCard;
        this.phoneNumber = phoneNumber;
        this.email = email;
        this.typeCustomer = typeCustomer;
        this.address = address;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public Service getService() {
        return service;
    }

    public void setService(Service service) {
        this.service = service;
    }

    public String getNameCustomer() {
        return nameCustomer;
    }

    public void setNameCustomer(String nameCustomer) {
        this.nameCustomer = nameCustomer;
    }

    public String getBirthday() {
        return birthday;
    }

    public void setBirthday(String birthday) {
        this.birthday = birthday;
    }

    public int getIdCard() {
        return idCard;
    }

    public void setIdCard(int idCard) {
        this.idCard = idCard;
    }

    public int getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(int phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getTypeCustomer() {
        return typeCustomer;
    }

    public void setTypeCustomer(String typeCustomer) {
        this.typeCustomer = typeCustomer;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }
    public String showInfo(){
        return "Name customer : "+this.getNameCustomer()+
                "\nGender : "+this.getGender()+
                "\nBirthday :" +this.getBirthday()+
                "\nId Card : "+this.getBirthday()+
                "\nPhone Number : "+this.getPhoneNumber()+
                "\nEmail : "+this.getEmail()+
                "\nType Customer : "+this.getTypeCustomer()+
                "\nAddress : "+this.getAddress();
    }
}
